### PR TITLE
CC | Integrate pause image inside rootfs

### PR DESCRIPTION
**Pull request by arronwy**
For CoCo stack, the pause image is managed by host side,
then it may configure a malicious pause image, we need package
a pause image inside the rootfs and don't the pause image from host.

The reasons we don't pull the pause image inside the guest, one is performance, another is the trusted storage for CoCo is not ready when launch the pause container https://github.com/confidential-containers/documentation/issues/39

Fixes: #4768

Signed-off-by: Wang, Arron <arron.wang@intel.com>

### CODE REVIEW

**Comment by wainersm** [comment]
@fidencio @arronwy hi, could you hold this PR until I finish my review?

File: tools/osbuilder/rootfs-builder/ubuntu/rootfs_lib.sh
Comment (wainersm) [inline tools/osbuilder/rootfs-builder/ubuntu/rootfs_lib.sh:46]: The `pause` image will be an important asset in the guest image; I suggest to put its url and version in the `versions.yaml` so that 1) can be easily bumped, 2) ensure same version/asset used on other distro guests

**Comment by wainersm** [comment]
@arronwy We will need to merge https://github.com/kata-containers/kata-containers/pull/4751 that picks up the fix to https://github.com/kata-containers/kata-containers/issues/4764 then you will need to rebase this PR. After all that, the CI should pass again :)

**Comment by arronwy** [comment]
> @arronwy We will need to merge #4751 that picks up the fix to #4764 then you will need to rebase this PR. After all that, the CI should pass again :)

Ok, I'll rebased this PR after #4751 is merged

File: tools/osbuilder/rootfs-builder/ubuntu/rootfs_lib.sh
Comment (arronwy) [reply tools/osbuilder/rootfs-builder/ubuntu/rootfs_lib.sh:46]: Thanks @wainersm , agree, I'll use `versions.yaml` to track `pause` image

**Comment by arronwy** [comment]
Rebased with latest CCv0 branch.

**Review by wainersm — APPROVED** [review]
@arronwy thanks for accepting my suggestions! LGTM.

**Comment by wainersm** [comment]
The failure on jenkins-ci-fedora-32-crio is related with this PR. It failed to build the ubuntu rootfs container:
```
/tmp/jenkins/workspace/kata-containers-CCv0-fedora-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder
+ podman build --runtime runc --build-arg http_proxy= --build-arg https_proxy= -t ubuntu-rootfs-osbuilder /tmp/jenkins/workspace/kata-containers-CCv0-fedora-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/ubuntu
[1/2] STEP 1/5: FROM golang:1.18 AS skopeo
[2/2] STEP 1/9: FROM docker.io/ubuntu:focal
Error: error creating build container: short-name resolution enforced but cannot prompt without a TTY
```
I never saw that error before...

**Comment by stevenhorsman** [comment]
> The failure on jenkins-ci-fedora-32-crio is related with this PR. It failed to build the ubuntu rootfs container:
> 
> ```
> /tmp/jenkins/workspace/kata-containers-CCv0-fedora-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder
> + podman build --runtime runc --build-arg http_proxy= --build-arg https_proxy= -t ubuntu-rootfs-osbuilder /tmp/jenkins/workspace/kata-containers-CCv0-fedora-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/ubuntu
> [1/2] STEP 1/5: FROM golang:1.18 AS skopeo
> [2/2] STEP 1/9: FROM docker.io/ubuntu:focal
> Error: error creating build container: short-name resolution enforced but cannot prompt without a TTY
> ```
> 
> I never saw that error before...

Would it be fair to guess that is related to podman if it just fails on the fedora run? (I think we use podman over docker only for fedora-like OS's?

**Comment by stevenhorsman** [comment]
> > The failure on jenkins-ci-fedora-32-crio is related with this PR. It failed to build the ubuntu rootfs container:
> > ```
> > /tmp/jenkins/workspace/kata-containers-CCv0-fedora-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder
> > + podman build --runtime runc --build-arg http_proxy= --build-arg https_proxy= -t ubuntu-rootfs-osbuilder /tmp/jenkins/workspace/kata-containers-CCv0-fedora-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/ubuntu
> > [1/2] STEP 1/5: FROM golang:1.18 AS skopeo
> > [2/2] STEP 1/9: FROM docker.io/ubuntu:focal
> > Error: error creating build container: short-name resolution enforced but cannot prompt without a TTY
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I never saw that error before...
> 
> Would it be fair to guess that is related to podman if it just fails on the fedora run? (I think we use podman over docker only for fedora-like OS's?

@arronwy @wainersm - I've had a little bit of time to look into it and I think the short version is that podman doesn't like using the golang:1.18 shortname, so if you try replacing it with `docker.io/library/golang:1.18` then I _think_ that should work.

**Comment by arronwy** [comment]
> > > The failure on jenkins-ci-fedora-32-crio is related with this PR. It failed to build the ubuntu rootfs container:
> > > ```
> > > /tmp/jenkins/workspace/kata-containers-CCv0-fedora-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder
> > > + podman build --runtime runc --build-arg http_proxy= --build-arg https_proxy= -t ubuntu-rootfs-osbuilder /tmp/jenkins/workspace/kata-containers-CCv0-fedora-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/ubuntu
> > > [1/2] STEP 1/5: FROM golang:1.18 AS skopeo
> > > [2/2] STEP 1/9: FROM docker.io/ubuntu:focal
> > > Error: error creating build container: short-name resolution enforced but cannot prompt without a TTY
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > I never saw that error before...
> > 
> > 
> > Would it be fair to guess that is related to podman if it just fails on the fedora run? (I think we use podman over docker only for fedora-like OS's?
> 
> @arronwy @wainersm - I've had a little bit of time to look into it and I think the short version is that podman doesn't like using the golang:1.18 shortname, so if you try replacing it with `docker.io/library/golang:1.18` then I _think_ that should work.

Thanks @stevenhorsman @wainersm , this should be the root casue, I verified in my local environment with podman, thanks a lot!

**Review by stevenhorsman — APPROVED** [review]
This looks good to me. Thanks Arron!

**Comment by stevenhorsman** [comment]
@fidencio - I'm not sure if you were planning to review this PR, but it has 2 approvals now and I think this blocks the secure storage PR? The make-kata-tarball build is failing, but we've seen that a bit recently across multiple PRs, so I'm not sure it is worth holding up for that? Obviously no rush if there are extra things you'd like to see/do on this, but if we are all good then I think we can get it merged?

**Review by fidencio — APPROVED** [review]
lgtm, @arronwy!

I'll try to get the test passing before merging, though.

**Pull request merged**
